offchain-modules/src/packages/handlers/ckb.ts

```diff
@@ -1,51 +1,340 @@
+      for (const tx of block.transactions) {
+        if (await this.isBurnTx(tx)) {
+          burnTxs.push(tx);
+        }
+      }
+      logger.debug('get new burn events and save to db', burnTxs);
+      if (burnTxs.length > 0) {
+        const ckbBurns = burnTxs.map((tx) => {
+          const recipientData = tx.outputsData[0].toString();
+          logger.debug(`amount: 0x${recipientData.slice(84, 116)}`);
+          logger.debug('amount: ', Amount.fromUInt128LE(`0x${recipientData.slice(84, 116)}`).toHexString());
+          const chain = Number(recipientData.slice(2, 4));
+          switch (chain) {
+            case ChainType.ETH:
+              return {
+                ckbTxHash: tx.hash,
+                asset: `0x${recipientData.slice(44, 84)}`,
+                chain,
+                amount: BigNumber.from(Amount.fromUInt128LE(`0x${recipientData.slice(84, 116)}`)).toHexString(),
+                recipientAddress: `0x${recipientData.slice(4, 44)}`,
+                blockNumber: latestHeight,
+              };
+            case ChainType.TRON:
+              return {
+                ckbTxHash: tx.hash,
+                asset: `0x${recipientData.slice(72, 78)}`,
+                chain,
+                amount: BigNumber.from(Amount.fromUInt128LE(`0x${recipientData.slice(78, 110)}`)).toHexString(),
+                recipientAddress: `0x${recipientData.slice(4, 72)}`,
+                blockNumber: latestHeight,
+              };
+            default:
+              throw new Error(`wrong burn chain type: ${chain}`);
+          }
+        });
+        await this.saveBurnEvent(ckbBurns);
+      }
+      latestHeight++;
+      await asyncSleep(1000);
+    }
+  }
+
+  async isBurnTx(tx: Transaction) {
+    if (tx.outputs.length < 1) {
+      return false;
+    }
+    const recipientData = tx.outputsData[0].toString();
+    logger.debug('recipientData:', recipientData);
+    // if (recipientData.length != 2 + 2 + 40 + 40 + 32 + 64 + 64) {
+    //   return false;
+    // }
+    let asset;
+    let assetAddress;
+    const chain = Number(recipientData.slice(2, 4));
+    switch (chain) {
+      case ChainType.ETH:
+        if (recipientData.length != 2 + 2 + 40 + 40 + 32 + 64 + 64) {
+          return false;
+        }
+        assetAddress = recipientData.slice(44, 84);
+        asset = new EthAsset(`0x${assetAddress}`);
+        break;
+      case ChainType.TRON:
+        if (recipientData.length != 2 + 2 + 68 + 6 + 32 + 64 + 64) {
+          return false;
+        }
+        assetAddress = recipientData.slice(72, 78);
+        asset = new TronAsset(uint8ArrayToString(fromHexString(assetAddress)));
+        logger.debug('tron asset: ', asset);
+        break;
+      default:
+        return false;
+      // throw new Error(`wrong burn chain type: ${chain}`);
+    }
+    // const asset = new EthAsset(recipient.asset);
+    const bridgeCellLockscript = {
+      codeHash: nconf.get('forceBridge:ckb:deps:bridgeLock:script:codeHash'),
```

Comment:
you can use `ForceBridgeCore.config.ckb.deps.bridgeLock.script.codeHash` to get it.

Reply:
fixed